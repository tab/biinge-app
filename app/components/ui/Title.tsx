import React from "react"
import { View, StyleProp, TextStyle } from "react-native"

import Typography from "components/ui/Typography"
import { titleStyles } from "styles"

type Props = {
  style?: StyleProp<TextStyle>
  aside?: React.ReactNode
  children: React.ReactNode
}

const TitleComponent = ({ style, aside, children }: Props) => {
  return (
    <View style={titleStyles.root}>
      <Typography
        variant="title1"
        style={[aside ? titleStyles.short : titleStyles.full, style]}
      >
        {children}
      </Typography>
      {aside && <>{aside}</>}
    </View>
  )
}

export default TitleComponent
