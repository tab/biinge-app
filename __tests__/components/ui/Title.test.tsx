import React from "react"
import "react-native"
import { Text } from "react-native"
import { render } from "@testing-library/react-native"
import { it } from "@jest/globals"

import Title from "components/ui/Title"

describe("Title", () => {
  const title = "Movie title"

  it("renders correctly", () => {
    const tree = render(<Title>{title}</Title>)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly with aside", () => {
    const tree = render(<Title aside={<Text>Aside</Text>}>{title}</Title>)
    expect(tree).toMatchSnapshot()
  })
})
