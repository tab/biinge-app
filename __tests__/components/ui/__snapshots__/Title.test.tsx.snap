// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "flexDirection": "row",
      "flexWrap": "nowrap",
      "justifyContent": "space-between",
      "marginBottom": 7,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      [
        {
          "color": "#181818",
          "fontSize": 26,
          "fontWeight": "700",
          "lineHeight": 28,
        },
        [
          {
            "flexBasis": "100%",
          },
          {
            "color": "#181818",
          },
          undefined,
        ],
      ]
    }
  >
    Movie title
  </Text>
</View>
`;

exports[`Title renders correctly with aside 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "flexDirection": "row",
      "flexWrap": "nowrap",
      "justifyContent": "space-between",
      "marginBottom": 7,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      [
        {
          "color": "#181818",
          "fontSize": 26,
          "fontWeight": "700",
          "lineHeight": 28,
        },
        [
          {
            "flexBasis": 635,
          },
          {
            "color": "#181818",
          },
          undefined,
        ],
      ]
    }
  >
    Movie title
  </Text>
  <Text>
    Aside
  </Text>
</View>
`;
